WalletConnect/web3inbox
# feat: refactor auth address prefix

# Description
- Add chain data to `w3iAuthFacade`
- React to chain switching in `internalAuthProvider` 
- Allow passing chain data to `externalAuthProvider`
- Update web3modal & its config
- **Preliminary list of chains, not final**
- Adapt signature modal trigger to work with chain switching
- Remove all hard coding of `eip155:1` (except for chat components, those will be fixed at a later date)
- Remove unreferenced/dead code that communicated with parent dapp (old `iframe` widget)
# Type of change

- [ ] Draft PR (breaking/non-breaking change which needs more work for having a proper functionality _[Mark this PR as ready to review only when completely ready]_)

# Fixes/Resolves (Optional)

Resolves #248 

# Checklist:


# Testing 

- Connect via WalletConnect using a mobile wallet
- Subscribe on ethereum chain to gm dapp
- Switch to polygon chain
- ensure there is no sub to gm dapp
- subscribe to gm dapp
- switch back to ethereum, ensure original gm dapp sub is there (via topic)
- switch to a different chain try again
